examples/rag_pipeline.py

```diff
@@ -212,6 +214,22 @@ async def init_and_query_es(self):
         answer = await engine.aquery(TRAVEL_QUESTION)
         self._print_query_result(answer)
 
+    async def use_HyDe(self):
+        """This example show how to use HyDE: HyDE enhances search results by generating Hypothetical doc(virtual
+        article), for more details please refer to the paper: http://arxiv.org/abs/2212.10496
+        Query Result:
+        Bob likes traveling.
+        """
+        self._print_title("Use HyDE to analysis query")
+        # 1.  save docs
+        engine = SimpleEngine.from_docs(input_files=[TRAVEL_DOC_PATH])
+        # create HyDE query engine
+        hyde_query_transformr = HyDEQueryTransformFactory().create_hyde_query_transform()
+        hyde_query_engine = TransformQueryEngine(engine, hyde_query_transformr)
```

Comment:
How to integrate with SimpleEngine, not directly TransformQueryEngine.
What I means is that one engine entrance to support  like query rewrite, rerank and so on.